scripts/get_target_distance_service.py

```diff
@@ -0,0 +1,116 @@
+        self.target_position = RobotTarget()                                            
+        self.isTargetSet = False      
+
+        rospy.Subscriber("/robot_target", RobotTarget, self.robot_target_callback)     
+        rospy.Subscriber("/robot_state", RobotState, self.robot_state_callback)
+        self.service = rospy.Service("get_target_distance", TargetDistance, \
+                                      self.handle_get_target_distance)
+                                      
+
+    def robot_target_callback(self, data):
+        # store the latest robot's target
+        self.target_position.x = data.x
+        self.target_position.y = data.y
+        # When first target arrives, set to True
+        if not self.isTargetSet:
+
+            self.isTargetSet = True
+
+    def robot_state_callback(self, data):
+        """Callback to robot_state subscriber, collects current position and values for av_speed"""
+        self.robot_current.x_pos = data.x_pos
+        self.robot_current.y_pos = data.y_pos
+        # av_speed window
+        if len(self.robot_speed_list) < self.av_speed_window_limit:
+            self.robot_speed_list.append((data.x_vel, data.y_vel))
+        elif len(self.robot_speed_list) == self.av_speed_window_limit:
+            # First remove the first element
+            self.robot_speed_list.pop(0)
+            # Add data at the end of the list
+            self.robot_speed_list.append((data.x_vel, data.y_vel))
+        else:
+            rospy.logerr("The list size is goes over the theshold.")
+            rospy.logerr("List size = %d, Threshold is = %d", len(self.robot_speed_list), self.av_speed_window_limit)
+
+    def handle_get_target_distance(self, req):
+        """
+        Handles the 'get_target_distance' service
+
+        Returns:
+        - GetDistanceToTargetResponse object, the response message containing calculated distance
+        and average speed.
+
+        If the target position is set, calculates the distance in x and y directions, the overall
+        distance as a line, and the average speed in x and y directions based on the robot's speed
+        history. 
+        If the target position is not set, returns zero values for distance, but returns the values
+        for average speed.
+        """
+        response = TargetDistanceResponse()
+        if self.isTargetSet:
+            # Calculate distance
+            response.dist_x = self.target_position.x - self.robot_current.x_pos
+            response.dist_y = self.target_position.y - self.robot_current.y_pos
+            rospy.loginfo("target_x = %d target_y %d", self.target_position.x, self.target_position.y)
+            # distance as a line
+            response.dist = sqrt((self.target_position.x - self.robot_current.x_pos)**2 + (self.target_position.y - self.robot_current.y_pos)**2)
+        else:
+            # There is no target, set distance to zero
+            response.dist_x = 0
+            response.dist_y = 0
```

Comment:
```suggestion
            response.dist_x = 0.0
            response.dist_y = 0.0
```
By setting `0` you are changing the number type in this field